cmd/agent-protocol-forwarder/main.go

```diff
@@ -29,6 +42,207 @@ type Config struct {
+		    "pod-name": "nginx-866fdb5bfb-b98nw",
+		    "tls-server-key": "-----BEGIN PRIVATE KEY-----\n....\n-----END PRIVATE KEY-----\n",
+		    "tls-server-cert": "-----BEGIN CERTIFICATE-----\n....\n-----END CERTIFICATE-----\n",
+		    "tls-client-ca": "-----BEGIN CERTIFICATE-----\n....\n-----END CERTIFICATE-----\n",
+			"aa-kbc-params": "cc_kbc::http://192.168.100.2:8080"
+
+		}
+	*/
+
+	// The response is base64 encoded
+
+	// Decode the base64 response
+	decoded, err := base64.StdEncoding.DecodeString(string(body))
+	if err != nil {
+		fmt.Printf("failed to decode b64 encoded userData: %s", err)
+		return ""
+	}
+
+	return string(decoded)
+}
+
+// Add method to parse userData and copy it to a file
+func parseUserData(userData string, dstFilePath string) error {
+
+	// Write userData to file specified in the dstFilePath var
+	// Create the directory and the file - /peerpod/daemon.json
+
+	// Split the dstFilePath into directory and file name
+	splitPath := strings.Split(dstFilePath, "/")
+	dir := strings.Join(splitPath[:len(splitPath)-1], "/")
+
+	// Create the directory.
+	err := os.MkdirAll(dir, 0755)
+	if err != nil {
+		return fmt.Errorf("failed to create directory: %s", err)
+
+	}
+
+	// Create the file
+	file, err := os.Create(dstFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to create file: %s", err)
+	}
+	defer file.Close()
+
+	// Write userData to file
+	_, err = file.WriteString(userData)
+	if err != nil {
+		return fmt.Errorf("failed to write userData to file: %s", err)
+	}
+
+	fmt.Printf("Wrote userData to file: %s\n", dstFilePath)
+
+	return nil
+
+}
+
+// Get daemon.Config from userData
+func getConfigFromUserData(userData string) daemon.Config {
+
+	// UnMarshal the userData into forwarder (daemon) Config struct
+	var daemonConfig daemon.Config
+
+	err := json.Unmarshal([]byte(userData), &daemonConfig)
+	if err != nil {
+		fmt.Printf("failed to unmarshal userData: %s", err)
+		return daemon.Config{}
+	}
+
+	return daemonConfig
+}
+
+// Add method to get the value of aa-kbc-param from userdata and replace the value of aa_kbc_params in the
+// /etc/agent-config.toml file
+func updateAAKBCParams(aaKBCParams string, agentConfigFile string) error {
+
+	// if aaKBCParams is empty then return
+	if aaKBCParams == "" || agentConfigFile == "" {
+		return fmt.Errorf("aaKBCParams or agentConfigFile is empty")
+	}
+	// Replace the aa_kbc_params line in agentConfigFile with the aaKBCParams value
+	// Read the agentConfigFile
+	agentConfig, err := os.ReadFile(agentConfigFile)
+	if err != nil {
+		return fmt.Errorf("failed to read agent config file: %s", err)
+	}
+
+	// Split the agentConfigFile into lines
+	lines := strings.Split(string(agentConfig), "\n")
+
+	// Loop through the lines and replace the line that starts with aa_kbc_params
+	for i, line := range lines {
+		if strings.Contains(line, "aa_kbc_params") {
+			lines[i] = fmt.Sprintf("aa_kbc_params = \"%s\"", aaKBCParams)
+			fmt.Printf("Updated line: %s\n", lines[i])
+		}
+	}
```

Comment:
good suggestion. I'll test it in the next iteration

Reply:
I have not included a toml parser yet since realised that changing the entire line with `aa_kbc_params` should suffice for now. 
Let me know if it's fine to use a toml parser as a future improvement?